core/src/test/scala/sttp/tapir/ShadowedEndpointCheckerTest.scala

```diff
@@ -0,0 +1,107 @@
+package sttp.tapir
+
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import sttp.tapir.util.ShadowedEndpointChecker
+
+class ShadowedEndpointCheckerTest extends AnyFlatSpecLike with Matchers {
+
+  it should "should detect shadowed endpoints with path variables that contain special characters" in {
+    val e1 = endpoint.get.in("[x]")
+    val e2 = endpoint.get.in("x")
+    val e3 = endpoint.get.in("/..." / paths)
+    val e4 = endpoint.get.in("x/")
+    val e5 = endpoint.get.in("x")
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3, e4, e5))
+
+    val expectedResult = List(ShadowedEndpoint(e5, e2))
+    result shouldBe expectedResult
+  }
+
+  it should "detect endpoint with wildcard shadowed by endpoint with fixed path" in {
+    val e1 = endpoint.get.in("x/y/z")
+    val e2 = endpoint.get.in("x" / "y" / "z")
+    val e3 = endpoint.get.in("x" / paths)
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3))
+
+    val expectedResult = List(ShadowedEndpoint(e3, e2))
```

Comment:
I don't think that's correct - no shadowing happens here. e2 is more specific, e3 is a catch-all. If they were swapped in the original list, shadowing would happen.